src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerGroupPackagesJsonHandler.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2018-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.composer.internal;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.group.GroupFacet;
+import org.sonatype.nexus.repository.group.GroupHandler;
+import org.sonatype.nexus.repository.http.HttpResponses;
+import org.sonatype.nexus.repository.http.HttpStatus;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.Response;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Handler for merging packages.json files together.
+ */
+@Named
+@Singleton
+public class ComposerGroupPackagesJsonHandler
+    extends GroupHandler
+{
+  private final ComposerJsonProcessor composerJsonProcessor;
+
+  @Inject
+  public ComposerGroupPackagesJsonHandler(final ComposerJsonProcessor composerJsonProcessor) {
+    this.composerJsonProcessor = checkNotNull(composerJsonProcessor);
+  }
+
+  @Override
+  protected Response doGet(@Nonnull final Context context,
```

Comment:
Trivial: I can't remember exactly but I thought we agreed that everything was considered non-null unless marked Nullable?

Reply:
> Trivial: I can't remember exactly but I thought we agreed that everything was considered non-null unless marked Nullable?

We did, but the superclass defines `@Nonnull` so IDEA complains that the annotation has been removed (as analysis of the code would think we have a lesser restriction on nullability here than in the parent class). Until we take that out I'd rather do this than get warnings from IDEs or from static analysis.